### PR TITLE
Switch to GitHub Actions

### LINKED ISSUE #528: Move CI to GitHub Actions
GitHub Actions seems like a great fit for this project, since we have tests across Linux, Mac, and Windows, and today that requires maintaining two different CI setups with Travis and Appveyor.

If someone wants to help us out with this, I'll be happy to work with you to get you the credentials you need to set it up! Hit me up on our [community slack](https://rust-bindings.slack.com/) (get your [auto-invite here](https://rust-bindings-slackin.herokuapp.com/))--my Slack handle is `dherman`.

### PR DESCRIPTION
**Pull request by lhr0909**
Resolves #528 

- [x] Run cargo test with a NodeJS version matrix and Rust toolchain matrix
- [x] Specific environment dependencies (apt dependencies etc)
- [x] Electron settings
- [x] Generate Documentation
- [x] Windows build
- [x] MacOS build

### CODE REVIEW

**Comment by lhr0909** [comment]
Currently build is being run on my fork: https://github.com/lhr0909/neon/actions

**Comment by amilajack** [comment]
Your CI seems to be [failing on linux builds](https://github.com/lhr0909/neon/actions/runs/142246164) because its timing out. Any ideas for why that might be?

**Comment by lhr0909** [comment]
https://github.com/lhr0909/neon/actions/runs/143132073 I think I resolved that @amilajack I believe it failed because of the electron set up wasn't there.

**Comment by lhr0909** [comment]
https://github.com/lhr0909/neon/actions/runs/163149424 docs workflow is ready to go now.

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Generate Docs
```
Comment (kjvalencik) [inline .github/workflows/docs.yml:1]: @dherman @amilajack Do we use Github Pages for docs anymore? I thought we switched to only using `docs.rs`?

**Comment by lhr0909** [comment]
https://github.com/lhr0909/neon/runs/854609942?check_suite_focus=true#step:6:388 I got very close to resolve the Windows build but seems like there is a 60 seconds timeout.

File: .github/workflows/docs.yml (hunk 1 above)
Comment (amilajack) [reply .github/workflows/docs.yml:1]: Yea we do. I think the domain is hosted with netlify. But our static content for the site is hosted with gh pages. Cargo generated API docs are hosted on `docs.rs`.

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+on: push
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [8.x, 10.x, 12.x]
```
Comment (amilajack) [inline .github/workflows/linux.yml:12]: We're okay with dropping support for node 8. We can change this to `[10.x, 12.x, 14.x]` and that might make this CI pass

**Comment by lhr0909** [comment]
https://github.com/lhr0909/neon/runs/854609942?check_suite_focus=true#step:6:416

https://github.com/lhr0909/neon/runs/862005882?check_suite_focus=true#step:6:417

Actually, after looking into the test further, seems like it is failing at the `tests::cli_test`, and it doesn't seem to be a consistent repro.

**Comment by dherman** [comment]
@lhr0909 I suspect this has to do with some recent Windows build problems (#406, #501), which we think are fixed with #537. I've reviewed #537 today and we should be able to merge it soon, and then let's see if that just makes these issues here go away. :)

**Comment by dherman** [comment]
Getting excited……………

anna-excited2

**Comment by lhr0909** [comment]
@dherman Thanks Dave! Let me know if we would like to do the MacOS workflow, I can try to set it up.

**Comment by dherman** [comment]
@lhr0909 A macOS workflow would be great, yes! 😄

**Comment by lhr0909** [comment]
https://github.com/lhr0909/neon/actions/runs/173784987 I just rebased on latest master and seems like the Windows test passed.

**Comment by lhr0909** [comment]
Tests pass.

**Comment by lhr0909** [comment]
I think I have removed Appveyor file from the branch, but not sure why it is still a check

**Comment by kjvalencik** [comment]
Appveyer will continue to build until it's merged.

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      with:
+        toolchain: stable
+    - name: Run cargo doc
+      run: cargo doc
+    - name: edit target/doc/index.html
+      run: 'echo "<meta http-equiv=refresh content=0;url=neon/index.html>" > target/doc/index.html'
+    - name: Push docs to gh-pages branch
+      uses: JamesIves/github-pages-deploy-action@releases/v3
+      with:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
[the remark this answers is not in the document]
Comment (kjvalencik) [reply .github/workflows/docs.yml:26]: GitHub automatically gives it to you. It has limited permissions and a short TTL. It's one of the best features of actions. ❤️

**Comment by amilajack** [comment]
Is it possible to consolidate these configs into a single config?

**Comment by lhr0909** [comment]
@amilajack I can make them into a single config, just that we will still end up having this long list of checks because of the node version matrix, rust toolchain versions, and platforms. Also, the conditional logic in a single config will be hard to maintain (for example, docs and `xvfb` setup on Linux for Electron test). Separate files are easier if we are going to see this long list of checks anyway.

File: .github/workflows/docs.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    - name: Use Rust ${{ matrix.rust-toolchain }}
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+    - name: Run cargo doc
+      run: cargo doc
+    - name: edit target/doc/index.html
+      run: 'echo "<meta http-equiv=refresh content=0;url=neon/index.html>" > target/doc/index.html'
+    - name: Push docs to gh-pages branch
+      uses: JamesIves/github-pages-deploy-action@releases/v3
```
Comment (kjvalencik) [inline .github/workflows/docs.yml:24]: Can we use the git hash on unofficial actions? Actions in the `actions` and `actions-rs` orgs should be good. Thanks!

```suggestion
      uses: JamesIves/github-pages-deploy-action@37acdb9c8dbc56e55255f36199ba2f3651b74db3
```

File: .github/workflows/linux.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: install g++-4.8
+      run: sudo apt-get install -y g++-4.8
+    - name: set DISPLAY env
+      run: echo "::set-env DISPLAY=:99.0"
+    - name: run cargo test
+      uses: GabrielBB/xvfb-action@v1.2
```
Comment (kjvalencik) [inline .github/workflows/linux.yml:30]:
```suggestion
      uses: GabrielBB/xvfb-action@e9adbdcd44108e54f12d6a49a373a074385b8512
```

**Comment by dherman** [comment]
@lhr0909 Thanks for updating! It looks like there are a couple of build failures. I think the Appveyor one doesn't matter since we'll disable it. Do the Actions failures matter or do you think we can merge?

**Comment by kjvalencik** [comment]
@dherman It looks like it might be a legitimate test failure on node 14 + windows. The others are cancelled builds because of that failed build.

**Comment by kjvalencik** [comment]
@dherman Looks like the failed build is the neon new bug that @amilajack's PR fixes.

I think this is good to merge.

**Comment by kjvalencik** [comment]
@lhr0909 Thanks for rh awesome contribution! I'm exited about this build matrix!

**Comment by dherman** [comment]
Thank you @lhr0909!!

**Pull request merged**